numpy/core/src/multiarray/dlpack.c

```diff
@@ -206,6 +206,12 @@ array_dlpack(PyArrayObject *self,
     if (data == NULL) {
         return NULL;
     }
+    if ((char *)PyArray_DATA(self) - data != 0) {
+        PyErr_SetString(PyExc_TypeError,
+                        "Offsets not clearly supported by this "
+                        "version of DLPack.");
+        return NULL;
+    }
```

Comment:
This code path can only be hit if we are re-wrapping a dlpack capsule with an offset. It is currently not tested, which should be done in a follow-on PR.

Reply:
Yeah, don't worry about it.  This is nonsense anyway, I deleted the whole thing (the logic was probably incomplete anyway) and replaced it with the following comment:
```
    /*
     * Note: the `dlpack.h` header suggests/standardizes that `data` must be
     * 256-byte aligned.  We ignore this intentionally, because `__dlpack__`
     * standardizes that `byte_offset` must be 0 (for now) to not break pytorch:
     * https://github.com/data-apis/array-api/issues/293#issuecomment-964111413
     *
     * We further assume that exporting fully unaligned data is OK even without
     * `byte_offset` since the standard does not reject it.
     * Presumably, pytorch will support importing `byte_offset != 0` and NumPy
     * can choose to use it starting about 2023.  At that point, it may be
     * that NumPy MUST use `byte_offset` to adhere to the standard (as
     * specified in the header)!
     */
```